packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -96,26 +96,97 @@ def test_metrics_containers(dcos_api_session):
     the statsd-emitter executor. When found, query it's /app endpoint to test that
     it's sending the statsd metrics as expected.
     """
+    # Helper func to check for non-unique CID's in a given /containers/id endpoint
+    def check_cid(registry):
+        if len(registry) <= 1:
+            return True
+
+        cid1 = registry[len(registry) - 1]
+        cid2 = registry[len(registry) - 2]
+        if cid1 != cid2:
+            raise ValueError('{} != {}'.format(cid1, cid2))
+
+        return True
+
     @retrying.retry(wait_fixed=2000, stop_max_delay=LATENCY * 1000)
     def test_containers(app_endpoints):
+
+        debug_eid = []
+
         for agent in app_endpoints:
             response = dcos_api_session.metrics.get('containers', node=agent.host)
+            assert response.status_code == 200, "got {}".format(
+                response.status_code)
+
             if len(response.json()) > 0:
                 for c in response.json():
-
                     # Test that /containers/<id> responds with expected data
-                    container_response = dcos_api_session.metrics.get('containers/{}'.format(c), node=agent.host)
-                    if container_response.status_code == 200 and 'executor_id' in container_response.json():
-                        # Executor ID value is "executor_id": "statsd-emitter.a094eed0-b017-11e6-a972-b2bcad3866cb"
-                        assert 'statsd-emitter' in container_response.json()['executor_id'].split('.'), 'statsd-emitter'
-                        ' was not found running on any slaves.'
+                    container_id_path = 'containers/{}'.format(c)
+                    container_response = dcos_api_session.metrics.get(container_id_path, node=agent.host)
+
+                    # /containers/<container_id> should always respond succesfully
+                    assert container_response.status_code == 200, 'got {}'.format(c)
+
+                    assert 'datapoints' in container_response.json(), 'got {}'.format(container_response.json())
+
+                    cid_registry = []
+                    for dp in container_response.json()['datapoints']:
+                        assert 'tags' in dp, 'got {}'.format(dp)
+                        assert len(dp['tags']) == 5, 'got {}'.format(
```

Comment:
what do you think about pulling this onto a single line now that it is so short?